### PR TITLE
Masonry: update state so it matches when items prop is updated

### PR DESCRIPTION
**Pull request by willyau**
This PR fixes: #318

### CODE REVIEW

**Comment by christianvuerings** [comment]
Deploy preview for *gestalt* ready!

Built with commit 9bfb356712ee02ecd2d5d6beaf34e0dc83c51b19

https://deploy-preview-319--gestalt.netlify.com

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  :memo: CHANGELOG - <i>Please add a CHANGELOG.md entry for your changes</i>
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  :file_folder: Bundlesize - <i>Bundle size increase detected, please review</i>
  </td>
    </tr>
  </tbody>
</table>


Filename | Size Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
gestalt.js | :small_red_triangle: 0.1% | :small_red_triangle: 0.1% | 267.79 KB | 268.1 KB | 60.06 KB | 60.1 KB
gestalt.es.js | :small_red_triangle: 0.1% | :small_red_triangle: 0.1% | 265.1 KB | 265.41 KB | 59.7 KB | 59.74 KB
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: packages/gestalt/src/Masonry.js [hunk 1]
```diff
@@ -129,6 +129,7 @@ function statesForRendering<T>(props: Props<T>, state: State<T>) {
   const { items } = state;
 
   // Full layout is possible
+  // $FlowFixMe https://github.com/facebook/flow/issues/6151
```
Comment (bishwei) [inline packages/gestalt/src/Masonry.js]: what are these flow errors and why weren't they there before?

Comment (willyau) [reply packages/gestalt/src/Masonry.js]: this is the exact flow error:
```
Missing type annotation for `T`.
```

not sure why it wasn't reported

**Pull request merged**